src/Pate/Verification/Simplify.hs

```diff
@@ -54,6 +57,47 @@ simplifySubPreds a = withValid $ withSym $ \sym -> do
       _ -> return e
   IO.withRunInIO $ \runInIO -> PEM.mapExpr sym (\e -> runInIO (simplifyPred' e)) a
 
+
+simplifyBVOps_trace ::
+  forall sym arch t solver fs tp.
+  sym ~ (W4B.ExprBuilder t solver fs) =>
+  sym ->
+  (forall tp'.
+     W4.SymExpr sym tp' ->
+     W4.SymExpr sym tp' ->
+     EquivM_ sym arch (W4.SymExpr sym tp')) ->
+  W4.SymExpr sym tp ->
+  EquivM sym arch (W4.SymExpr sym tp)
+simplifyBVOps_trace sym checkWork outer = do
+  cache <- W4B.newIdxCache
+  let
+    go :: forall tp'. W4.SymExpr sym tp' -> EquivM_ sym arch (W4.SymExpr sym tp')
+    go e = W4B.idxCacheEval cache e $ WEH.simplifyBVOps' sym checkWork e
+  go outer
+
+simplifyWithSolver ::
```

Comment:
What does this use the solver for? What is the behavior if there is a timeout?

Reply:
Added a comment explaining this